frontend/src/pages/modelRegistrySettings/ModelRegistryDatabasePassword.tsx

```diff
@@ -1,23 +1,70 @@
 import React from 'react';
-import { Alert } from '@patternfly/react-core';
-import PasswordHiddenText from '~/components/PasswordHiddenText';
+import { Alert, Skeleton } from '@patternfly/react-core';
+import PasswordInput from '~/components/PasswordInput';
+import { ModelRegistryKind } from '~/k8sTypes';
+import useFetchState from '~/utilities/useFetchState';
+import { getModelRegistryBackend } from '~/services/modelRegistrySettingsService';
 
 type ModelRegistryDatabasePasswordProps = {
-  password?: string;
-  loadError?: Error;
+  password: string | undefined;
+  setPassword: (value: string) => void;
+  showPassword?: boolean;
+  isPasswordTouched?: boolean;
+  setIsPasswordTouched: (value: boolean) => void;
+  editRegistry: ModelRegistryKind;
 };
 
 const ModelRegistryDatabasePassword: React.FC<ModelRegistryDatabasePasswordProps> = ({
-  password,
-  loadError,
+  password = '',
+  setPassword,
+  showPassword,
+  isPasswordTouched,
+  setIsPasswordTouched,
+  editRegistry: mr,
 }) => {
-  if (loadError) {
-    return <Alert variant="danger" isInline isPlain title="Error loading password" />;
+  const [dbPassword, passwordLoaded, passwordLoadError] = useFetchState(
+    React.useCallback(async () => {
+      const { databasePassword } = await getModelRegistryBackend(mr.metadata.name);
+      return databasePassword;
+    }, [mr]),
+    undefined,
+  );
+
+  React.useEffect(() => {
+    if (dbPassword) {
+      setPassword(dbPassword);
+    }
+  }, [dbPassword, setPassword]);
+
+  const hasContent = (value: string): boolean => !!value.trim().length;
+
+  if (!passwordLoaded && !passwordLoadError) {
```

Comment:
This condition could also have `mr` in it to only apply to the editing case, but I think the NotReadyError may cover us there